tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/visibility/retry/RetryVisibilityTest.java

```diff
@@ -272,11 +273,11 @@ private void checkServiceCall(int nbExpectedRetries, RetryService service, Strin
                             RetryVisibilityTest.class.getSimpleName(),
                             testName,
                             expectedNbCalls));
-        } catch (RuntimeException ex) {
+        } catch (TestException ex) {
```

Comment:
Most changes in this class should be undone.

This class is using `IOException` as its expected exception - similar to the way we're using `TestException` in other classes.

If anything, this catch block could change from `RuntimeException` to just `Exception` - it's just adding more information about unexpected thrown exceptions.

You _could_ change `IOException` to `TestException` in this class and in `BaseRetryOnClassService`.